helper/helper_test.go

```diff
@@ -98,4 +295,234 @@ var _ = Describe("helper tests", func() {
 			Expect(isSubset).To(Equal(false))
 		})
 	})
+
+	Describe("doRestoreAgent Mocked unit tests", func() {
+		BeforeEach(func() {
+			// Setup mocked tests environment
+			*singleDataFile = false
+			*content = 1
+			*oidFile = "testoid.dat"
+			*isResizeRestore = true
+			*origSize = 5
+			*destSize = 3
+			*pipeFile = "mock"
+			*onErrorContinue = true
+		})
+		It("successfully restores using a single data file when inputs are valid and no errors occur", func() {
+			*singleDataFile = true
+
+			oidBatch := []oidWithBatch{{oid: 1, batch: 1}}
+			steps := []helperTestStep{
+				{"mock_1_1", true, 1, false, "Can open single data file"},
+			}
+
+			mockHelper := newHelperTest(oidBatch, steps)
+
+			// Prepare and write the toc file
+			testDir := "" // Use local directory for the TOC file instead of default
+			*tocFile = fmt.Sprintf("%stest_toc.yaml", testDir)
+			writeTestTOC(*tocFile)
+			defer func() {
+				_ = os.Remove(*tocFile)
```

Comment:
what if `writeTestTOC` does not create file?

Reply:
`writeTestTOC` has  
```
	fToc, err := os.Create(tocFile)
	Expect(err).ShouldNot(HaveOccurred())
```, then it will fail inside `writeTestTOC`